ReSwiftDemo/Services/SessionManager.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import ReSwift
+
+class SessionManager: StoreSubscriber {
+    
+    private let store: Store<AppState>
+    private var token: String?
+	
+    init(store: Store<AppState>) {
+        self.store = store
+        self.store.subscribe(self)
+    }
+    
+    func loadSession() -> (String, User)? {
+        self.token = UserDefaults.standard.string(forKey: "token")
+        
+        if let token = self.token,
+            let data = UserDefaults.standard.value(forKey: "user") as? Data,
+            let user = try? PropertyListDecoder().decode(User.self, from: data) {
+            
+            return (token, user)
+        }
+        
+        return nil
+    }
+    
+    func newState(state: AppState) {
+        guard state.flowState.flow != .splashScreen else { return }
+        guard self.token != state.sessionState.token else { return }
+        
+        // warning: if you quickly terminate app after these operations, UserDefaults might not be up to date
+        
+        if state.sessionState.token != nil {
```

Comment:
jak bys przetestlwal ta logike? ;)

Reply:
W sensie chodzi Ci że UserDefaults jest problematyczne?

Reply:
Dokładnie tak :) Szczególnie, że jak będziesz mieć różne testy, które z tego korzystają to po pewnym czasie będziesz mieć w tym straszny śmietnik :)